### PR TITLE
fix: add empty judgement to index search

### PR DESCRIPTION
**Pull request by AnneYang720**
This PR is related to issue #1530 .

### CODE REVIEW

**Review by JoanFM — CHANGES_REQUESTED** [review]
can we check what happens with other backends?

**Review by JoanFM — APPROVED** [review]

File: docarray/index/backends/hnswlib.py [hunk 1]
```diff
@@ -280,6 +280,9 @@ def _find_batched(
         limit: int,
         search_field: str = '',
     ) -> _FindResultBatched:
+        if not self.num_docs():
```
Comment (JoanFM) [inline docarray/index/backends/hnswlib.py:283]: do == 0

File: docarray/index/backends/in_memory.py [hunk 2]
```diff
@@ -194,7 +194,7 @@ def find(
         self._logger.debug(f'Executing `find` for search field {search_field}')
         self._validate_search_field(search_field)
 
-        if not len(self._docs):
+        if not self.num_docs():
```
Comment (JoanFM) [inline docarray/index/backends/in_memory.py:197]: can u do ==0?

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Review by JoanFM — APPROVED** [review]

**Comment by jupyterjazz** [comment]
@JoanFM  only hnsw failed when the index was empty, others are doing well

**Pull request merged**